api/backend.go

```diff
@@ -228,8 +227,22 @@ func (b *StatusBackend) CallPrivateRPC(inputJSON string) string {
 }
 
 // SendTransaction creates a new transaction and waits until it's complete.
-func (b *StatusBackend) SendTransaction(ctx context.Context, args transactions.SendTxArgs) (hash gethcommon.Hash, err error) {
-	return b.transactor.SendTransaction(ctx, args)
+func (b *StatusBackend) SendTransaction(sendArgs transactions.SendTxArgs, password string) sign.Result {
+	verifiedAccount, err := b.getVerifiedAccount(password)
+	if err != nil {
+		return sign.NewErrResult(err)
+	}
+	return b.transactor.SendTransaction(sendArgs, verifiedAccount)
+}
+
+// SignMessage checks the pwd vs the selected account and passes on the metadata
+// to personalAPI for message signature
+func (b *StatusBackend) SignMessage(rpcParams personal.Metadata) sign.Result {
+	_, err := b.getVerifiedAccount(rpcParams.Password)
+	if err != nil {
+		return sign.NewErrResult(err)
+	}
+	return b.personalAPI.Sign(context.Background(), rpcParams)
```

Comment:
@adambabik what is the right context to pass onto sign here? Couldn't find any calls to Sign before this PR so I don't have a reference of which one needs to be used.

Sign uses the context to pass along to api.rpcClient.CallContextIgnoringLocalHandlers

Reply:
This is a little bit messy because `transactions.Transactor` has a field with a timeout. We can add one to `personal.PublicAPI` and remove passing context to `PublicAPI.Sign()` or keep it but wrap it in `PublicAPI.Sign()` implementation:
```go
func (api *PublicAPI) Sign(ctx context.Context, rpcParams Metadata) sign.Result {
  response := sign.EmptyResponse
  ctx, cancel := context.WithTimeout(ctx, api.rpcCallTimeout)
  defer cancel()
  err := api.rpcClient.CallContextIgnoringLocalHandlers(
  ctx,
  &response,
  params.PersonalSignMethodName,
  rpcParams.Data, rpcParams.Address, rpcParams.Password)
  // ...
}
```